pio/ws2812/ws2812_parallel.c

```diff
@@ -16,9 +16,11 @@
 #include "ws2812.pio.h"
 
 #define FRAC_BITS 4
-#define PIN_TX 2
-// make sure that this number is less or equal to the MAX_LENGTH defined below
 #define NUM_PIXELS 64
+#ifndef PICO_DEFAULT_WS2812_PIN
+#warning "no WS2812 default PIN defined for board, please check if pin 2 is okay"
+#define PICO_DEFAULT_WS2812_PIN 2
```

Comment:
While the change to `ws2812.c` seems okay, I'm not sure if the change here is appropriate?
`ws2812_parallel_program_init` in `ws2812.pio` says:
```
static inline void ws2812_parallel_program_init(PIO pio, uint sm, uint offset, uint pin_base, uint pin_count, float freq) {
    for(uint i=pin_base; i<pin_base+pin_count; i++) {
        pio_gpio_init(pio, i);
    }
```
i.e. this `ws2812_parallel.c` example actually uses both pins 2 and 3. Probably makes sense to put it back to using `PIN_TX`, but also rename it to `PIN_TX_BASE` to make things more obvious?

Reply:
That makes sense. I renamed it to `WS2812_PIN_BASE`.